gossipd/routing.c

```diff
@@ -1685,5 +1689,6 @@ void handle_local_add_channel(struct routing_state *rstate, const u8 *msg)
 		     type_to_string(tmpctx, struct short_channel_id, &scid));
 
 	/* Create new (unannounced) channel */
-	new_chan(rstate, &scid, &rstate->local_id, &remote_node_id);
+	chan = new_chan(rstate, &scid, &rstate->local_id, &remote_node_id);
+	chan->satoshis = satoshis;
```

Comment:
We should move the satoshis setting into new_chan, since it's always set immediately now...